pontoon/base/views.py

```diff
@@ -348,6 +348,93 @@ def get_translation_history(request):
     return JsonResponse(payload, safe=False)
 
 
+@utils.require_AJAX
+@login_required(redirect_field_name='', login_url='/403')
+@transaction.atomic
+def approve_translation(request):
+    """Approve given translation."""
+    try:
+        t = request.POST['translation']
+        paths = request.POST.getlist('paths[]')
+    except MultiValueDictKeyError as e:
+        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
+
+    translation = get_object_or_404(Translation, pk=t)
+    project = translation.entity.resource.project
+    locale = translation.locale
+    user = request.user
+
+    # Read-only translations cannot be approved
+    if utils.readonly_exists(project, locale):
+        return HttpResponseForbidden(
+            "Forbidden: This translation is in read-only mode"
+        )
+
+    if translation.approved:
+        return HttpResponseForbidden(
+            "Forbidden: This translation is already approved"
+        )
+
+    # Only privileged users can approve translations
+    if not user.can_translate(locale, project):
+        return HttpResponseForbidden(
+            "Forbidden: You don't have permission to approve this translation."
+        )
+
+    # Check for errors.
+    if project.slug == 'tutorial':
+        # Checks are disabled for the tutorial.
+        use_ttk_checks = False
+    else:
+        try:
+            use_ttk_checks = UserProfile.objects.get(user=user).quality_checks
+        except UserProfile.DoesNotExist:
+            use_ttk_checks = True
+
+    if use_ttk_checks and translation.errors.all():
```

Comment:
Hey :)
1. From what I remember (@mathjazz and @pike can correct me if something has changed in that matter) Pontoon can't allow the users to approve translations with errors.
https://github.com/mozilla/pontoon/blob/master/pontoon/checks/utils.py#L94

So, this line should look like this:
```python
if translation.errors.exist():
```
2. Translations can have warnings and the translator can choose to ignore them during the approval process.

Reply:
Agreed. We should use `are_blocking_checks` here.